docker-compose.yml

```diff
@@ -86,84 +160,177 @@ services:
+      #     BUNDLER_VERSION_OVERRIDE: "2.2.14"
+      #     NODE_MAJOR: "12"
+      #     YARN_VERSION: "1.22.4"
     image: medicaid_gateway:dev-0.1.0
     volumes:
       - ../medicaid_gateway:/medicaid_gateway
-      - gem_cache:/usr/local/bundle/2.7.2
+      # - mcg_gem_cache:/usr/local/bundle/2.7
+      # - mcg_rails_cache:/medicaid_gateway/tmp/cache
+      #- .:/medicaid_gateway:cached
+      - mcg_rails_cache:/medicaid_gateway/tmp/cache
+      - mcg_bundle:/bundle
+      - mcg_node_modules:/medicaid_gateway/node_modules
+      - mcg_packs:/medicaid_gateway/public/packs
     env_file:
-      - ./site/.env
-      - ./site/rabbitmq/rabbitmq.env
-      - ./site/medicaid_gateway_service/.env
+      - ./site/.env.dev
+      - ./site/rabbitmq/rabbitmq.dev.env
+      - ./site/medicaid_gateway_service/.env.dev
     depends_on:
+      medicaid_gateway:
+        condition: service_healthy
+    links:
+      - mitc
       - mongodb
       - rabbitmq
       - redis
+    command: bash -c "bundle exec rake configuration:client_configuration_toggler client='dc' && bin/webpack"
+
+
+  medicaid_gateway:
+    build:
+      context: ../medicaid_gateway
+      dockerfile: ../ea_enterprise/site/medicaid_gateway_service/Dockerfile
+      #dockerfile: ../medicaid_gateway/.docker/development/Dockerfile
+      # args:
+      #     RUBY_VERSION: "2.7.2"
+      #     BUNDLER_VERSION_OVERRIDE: "2.2.14"
+      #     NODE_MAJOR: "12"
+      #     YARN_VERSION: "1.22.4"
+    image: medicaid_gateway:dev-0.1.0
+    volumes:
+      - ../medicaid_gateway:/medicaid_gateway
+      # - mcg_gem_cache:/usr/local/bundle/2.7
+      # - mcg_rails_cache:/medicaid_gateway/tmp/cache
+      #- .:/medicaid_gateway:cached
+      - mcg_rails_cache:/medicaid_gateway/tmp/cache
+      - mcg_bundle:/bundle
+      - mcg_node_modules:/medicaid_gateway/node_modules
+      - mcg_packs:/medicaid_gateway/public/packs
+      - ../aca_entities:/aca_entities
+    env_file:
+      - ./site/.env.dev
+      - ./site/rabbitmq/rabbitmq.dev.env
+      - ./site/medicaid_gateway_service/.env.dev
+      - env.dev
+    depends_on:
+      mongodb:
+        condition: service_started
+      rabbitmq:
+        condition: service_healthy
+      redis:
+        condition: service_healthy
     links:
       - mitc
+      - mongodb
+      - rabbitmq
+      - redis
     ports:
       - "3005:3005"
-    command: bash -c "rm -f tmp/pids/server.pid && mkdir -p tmp/pids && bundle install && bin/rails server --port 3005 --binding 0.0.0.0"
-
+    healthcheck:
+      test: curl --fail http://localhost:3005 || exit 1
+      start_period: 20s 
+      interval: 3m
+    #this works for sure
+    # command: bash -c "rm -rf tmp/pids/server.pid && mkdir -p tmp/pids && bundle install && yarn install --check-files && bin/rails server --port 3005 --binding 0.0.0.0"
+    command: bash -c "rm -rf tmp/pids/server.pid && mkdir -p tmp/pids && bin/rails dev:cache && bin/rails server --port 3005 --binding 0.0.0.0"
```

Comment:
I think including `dev:cache` here will cause problems because it will toggle caching in development on/off every time the container is started (and Stimulus Reflex requires it to be enabled for the app to start).  

I see this command is also included in site/medicaid_gateway_service/Dockerfile, which may be sufficient to enable it once when the image is built.  Need to confirm this though.